src/main/java/duke/main/Parser.java

```diff
@@ -0,0 +1,124 @@
+package duke.main;
+
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.TodoCommand;
+
+import duke.exception.DeadlineIncompleteException;
+import duke.exception.DeleteIncompleteException;
+import duke.exception.DeleteOutOfListException;
+import duke.exception.DoneIncompleteException;
+import duke.exception.DoneOutOfListException;
+import duke.exception.DukeException;
+import duke.exception.EventIncompleteException;
+import duke.exception.FindIncompleteException;
+import duke.exception.NoInputException;
+import duke.exception.TodoIncompleteException;
+import duke.exception.UnknownInputException;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/**
+ * Parser is used to make sense the input from the user and convert it to executable command.
+ */
+public class Parser {
+
+    public static String[] commands = {"list", "done", "deadline", "event", "todo", "delete", "bye", "find"};
+    public static ArrayList<String> VALID_COMMAND = new ArrayList<>(Arrays.asList(commands));
+
+    /**
+     * Checks the correctness of the input by the user.
+     *
+     * @param line The input from the user.
+     * @throws DukeException If user's input is invalid.
+     */
+    public static void checkInput(String line) throws DukeException {
+        String[] input = line.split(" ");
+        String command = input[0];
+        if (command.equals("")) {
+            throw new NoInputException();
+        } else if (!VALID_COMMAND.contains(input[0])) {
+            throw new UnknownInputException();
+        } else if (input.length == 1) {
+            switch (command) {
```

Comment:
Should the "case"s be on the same indentation as "switch"?

Reply:
Oh yes, I think I overlook the indentation because it was automated by the IDE (and I haven't changed the styling setting yet) 😅